kernel/src/tbfheader.rs

```diff
@@ -268,35 +187,6 @@ crate unsafe fn parse_and_validate_tbf_header(address: *const u8) -> Option<TbfH
     let version = *(address as *const u16);
 
     match version {
-        1 => {
```

Comment:
Not that I think anyone is still using v1, but on the off chance, maybe a panic on this match arm with something like "An app has a TBF version 1 header. This is no longer supported by Tock. Please update and recompile your apps."

Reply:
I considered that, but then a bad app can crash the system, which isn't what we want.

Reply:
Mhmm... good point. How about `debug!("Ignoring app with TBF v1 header")` -- fighting chance for anyone struggling but unlikely to show up for any normal user

Reply:
Can't `debug!()` cause a panic as well, if the board designer isn't expecting there to be any debug statements (and doesn't initialize it), and this is "hiding" in the kernel? Or will it somehow get optimized out?

Reply:
> but then a bad app can crash the system, which isn't what we want.

well... the app itself can't, the _header_ can. which is kind of can anyway---I think of the process header as being trusted, no? (obviously better if it doesn't have to be)

In any case, I don't think this library should be printing debug statements or panicing. If we need a more specific differentiation of the error, it should be reflected in the return type (e.g. a `Result<TbfHeader, TbfError>` instead of an `Option<TbfHEader>` or something).

I think just returning None for now is reasonable, though.